### PR TITLE
DT-2215 Use the queue lib's endopints for retry and purge

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PrisonerIndexResourceTest.kt [hunk 1]
```diff
@@ -673,3 +679,44 @@ class PrisonerIndexResourceTest : QueueIntegrationTest() {
 }
 
 private fun String.readResourceAsText(): String = PrisonerIndexResourceTest::class.java.getResource(this).readText()
+
+fun populateOffenderMessage(offenderNumber: String) =
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PrisonerIndexResourceTest.kt:683]: Moved from a deleted class.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PrisonerIndexResource.kt [hunk 2]
```diff
@@ -157,7 +99,5 @@ class PrisonerIndexResource(
     ]
   ) fun indexQueueHousekeeping() {
     prisonerIndexService.indexingComplete(ignoreThreshold = false)
-    queueAdminService.transferIndexMessages()
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PrisonerIndexResource.kt:160]: not quite sure where these have gone too?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/config/JmsLocalStackConfig.kt [hunk 3]
```diff
@@ -12,19 +12,30 @@ import org.springframework.beans.factory.annotation.Value
...
+    awsSqsDlqClient: AmazonSQS,
+    @Value("\${sqs.dlq.name}") dlqName: String,
   ): AmazonSQS =
     AmazonSQSClientBuilder.standard()
       .withEndpointConfiguration(AwsClientBuilder.EndpointConfiguration(serviceEndpoint, region))
       .withCredentials(AWSStaticCredentialsProvider(AnonymousAWSCredentials()))
       .build()
+      .also {
+        hmppsQueueService.registerHmppsQueue(
+          HmppsQueue(it, queueName, awsSqsDlqClient, dlqName)
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/config/JmsLocalStackConfig.kt:36]: I know I haven't commented on this before (sorry), but this method looks odd on the queue service.  In kotlin land seems odd to have to call the constructor on a queue and then register it.  Might be nicer to just call the queue (by parameter names):
```
hmppsQueueService.registerHmppsQueue(client = it, queue = queueName, dlqClient = awsSqsDlqClient, dlqQueue = dlqName)
```

**Review by petergphillips — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PrisonerIndexResource.kt (hunk 2 above)
Comment (mikehalmamoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PrisonerIndexResource.kt:160]: These are replaced by the `/queue-admin/retry-all-dlqs` endpoint from the queue lib.

**Pull request merged**